actionview/lib/action_view/helpers/url_helper.rb

```diff
@@ -542,7 +542,7 @@ def current_page?(options, check_parameters: false)
 
         return false unless request.get? || request.head?
 
-        check_parameters ||= !options.is_a?(String) && options.try(:delete, :check_parameters)
+        check_parameters ||= options.respond_to?(:to_hash) && options.try(:delete, :check_parameters)
```

Comment:
Maybe just change the type check to `is_a?(Hash)` instead? (And no need for the `try`)

That way we can keep the kwarg, which seems better for documentation.